backend/open_webui/routers/retrieval.py

```diff
@@ -2313,6 +2212,223 @@ async def cleanup_expired_web_searches(
         return {"error": str(e), "collections_cleaned": 0, "vectors_cleaned": 0}
 
 
+async def cleanup_expired_web_searches_safe(
+    max_age_days: int = 30,
+    force_delete_all: bool = False,
+    batch_size: int = 10,
+    max_concurrent: int = 5,
+):
+    """
+    K8s-safe cleanup of expired web search collections with batching and concurrency control.
+
+    Features:
+    - Batched processing to prevent memory spikes
+    - Controlled concurrency to prevent database overload
+    - Non-blocking with regular yields to prevent pod lockup
+    - Timeout protection
+    - Error isolation (one failure doesn't stop others)
+
+    Args:
+        max_age_days: Age threshold for deletion
+        force_delete_all: Delete all web collections regardless of age
+        batch_size: Number of collections to process in each batch
+        max_concurrent: Maximum concurrent deletion operations
+    """
+    import time
+
+    start_time = time.time()
+
+    try:
+        cleanup_summary = {
+            "collections_checked": 0,
+            "collections_cleaned": 0,
+            "vectors_cleaned": 0,
+            "errors": [],
+            "batches_processed": 0,
+            "start_time": start_time,
+        }
+
+        if not VECTOR_DB_CLIENT:
+            cleanup_summary["errors"].append("Vector DB client not available")
+            return cleanup_summary
+
+        log.info(
+            f"🧹 Starting K8s-safe web search cleanup (batch_size={batch_size}, max_concurrent={max_concurrent})"
+        )
+
+        # Get collections with timeout protection
+        collections = []
+        try:
+            async with asyncio.timeout(30):  # 30 second timeout for collection listing
```

Comment:
This should be fine for qdrant! It's able to send all of the collections in Prod in `"time": 0.003521335`